app/controllers/tasks.js

```diff
@@ -1,67 +1,61 @@
 import Controller from '@ember/controller';
 import { tracked } from '@glimmer/tracking';
-import { action, set } from '@ember/object';
+import { action } from '@ember/object';
 import ENV from 'website-my/config/environment';
-import TASK_STATUS from 'website-my/constants/tasks';
+import { TASK_STATUSES } from 'website-my/constants/tasks';
 
 const API_BASE_URL = ENV.BASE_API_URL;
-const { ACTIVE, BLOCKED, COMPLETED, PENDING } = TASK_STATUS;
 
 export default class TasksController extends Controller {
-  @tracked showDropDown = false;
+  taskStatuses = TASK_STATUSES;
+  DEFAULT_TASK_TYPE = this.taskStatuses[0];
+  @tracked showDropDown = true;
   @tracked taskFields = {};
   @tracked allTasks = this.model;
+  @tracked isLoading = false;
+  @tracked userSelectedTask = this.DEFAULT_TASK_TYPE;
 
   @action toggleDropDown() {
     this.showDropDown = !this.showDropDown;
   }
 
   @tracked tasksByStatus = {};
-  taskStatusList = [ACTIVE, BLOCKED, PENDING];
 
-  filterTasksByStatus(allTasks, status) {
-    return allTasks.filter((task) => task.status === status);
-  }
-
-  seperateTasksByStatus() {
-    this.taskStatusList.forEach((status) => {
-      this.tasksByStatus[status] = this.filterTasksByStatus(
-        this.allTasks,
-        status
+  calculateTasksToShowBySelectedStatus() {
```

Comment:
Would this be better as a `get`?

Reply:
What I mean is that here, `calculateTasksToShowBySelectedStatus` is being called imperatively. From what I see in its implementation, it is used to update the details of `tasksToShow`, which in turns depends on the `userSelectedTask.key`

We should be only setting the `userSelectedTask.key`, and have `tasksToShow` be 'derived' from the dependency(s). So, we derive the data, instead of having to remember to call the `calculateTasksToShowBySelectedStatus`.

Read more on[ tracked properties](https://guides.emberjs.com/release/upgrading/current-edition/tracked-properties/) and let me know if you still have any questions.

Reply:
We've added functionality to view all tasks (no filtering based on task status)
Here's the function definition:
```JS
  filterTasksByStatus() {
    if (this.userSelectedTask.key === this.allTasksObject.key) {
      this.tasksToShow = this.allTasks;
    } else {
      this.tasksToShow = this.allTasks.filter(
        (task) => task.status === this.userSelectedTask.key
      );
    }
  }
```